google-cloud-storage/src/main/java/com/google/cloud/storage/testing/StorageRpcTestBase.java

```diff
@@ -0,0 +1,505 @@
+   * UnsupportedOperationException}, depending on the {@code isImplemented} parameter.
+   *
+   * @param isImplemented {@code true} to return a value, {@code false} to throw an exception.
+   */
+  public StorageRpcTestBase(boolean isImplemented) {
+    this.isImplemented = isImplemented;
+  }
+
+  @Override
+  public Bucket create(Bucket bucket, Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public StorageObject create(StorageObject object, InputStream content, Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<Bucket>> list(Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<StorageObject>> list(String bucket, Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public Bucket get(Bucket bucket, Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public StorageObject get(StorageObject object, Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public Bucket patch(Bucket bucket, Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public StorageObject patch(StorageObject storageObject, Map<Option, ?> options) {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public boolean delete(Bucket bucket, Map<Option, ?> options) {
+    if (isImplemented) {
+      return false;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public boolean delete(StorageObject object, Map<Option, ?> options) {
+    if (isImplemented) {
+      return false;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public RpcBatch createBatch() {
+    if (isImplemented) {
+      return null;
+    } else {
+      throw new UnsupportedOperationException("Not implemented yet");
+    }
+  }
+
+  @Override
+  public StorageObject compose(
+      Iterable<StorageObject> sources, StorageObject target, Map<Option, ?> targetOptions) {
+    if (isImplemented) {
```

Comment:
Why is "isImplemented" needed? If you're just returning null here and elsewhere and not expecting these methods to be called, unconditionally throwing the exception might be better.

Reply:
Yes, that was my original intention to throw the exception unconditionally. Then I decided that adding "isImplemented" feature might simplify the work on extending that class.

Reply:
It doesn't seem like returning null helps a lot in this case. Subclassers who invoke a method like this will still need to override this.

Reply:
I tend to agree with you, but I will think more about cases when returning null is useful.

Reply:
Fixed as you suggested. If someone needs an implementation returning null, an IDE can help to produce such class.